src/IOHprofiler_csv_logger.cpp

```diff
@@ -0,0 +1,142 @@
+#include "IOHprofiler_csv_logger.h"
+
+int IOHprofiler_csv_logger::openIndex() { 
+  std::string experiment_folder_name = IOHprofiler_experiment_folder_name();
+  IOHprofiler_create_folder(experiment_folder_name);
+}
+
+
+int IOHprofiler_csv_logger::IOHprofiler_create_folder(std::string folder_name) { 
+  if(fs::create_directory(folder_name)) {
+    return 1;
+  } else {
+    IOH_error("Error on creating directory");
+    return 0;
+  }
+}
+
+
+// To create a folder for logging files.
+// If there exists a files or a folder with the same name, the expected 
+// directory will be renamed by adding a suffix.
+// For example,
+//     If a folder or file 'test' has already been in currect path, the 
+//     expected directory will be renamed as 'test-1', 'test-2', ... 
+//     until there is no such a folder or file. 
+std::string IOHprofiler_csv_logger::IOHprofiler_experiment_folder_name() {
+  std::string renamed_directory = this->output_directory + IOHprofiler_path_separator + this->folder_name;
+  std::string temp_folder_name;
+  int index = 0;
+  while(fs::exists(renamed_directory.c_str())) {
+    ++index;
+    temp_folder_name = this->folder_name  + '-' + std::to_string(index);
+    renamed_directory = this->output_directory + IOHprofiler_path_separator + temp_folder_name;
+  }
+  this->folder_name = temp_folder_name;
+  return renamed_directory;
+}
+
+
+void IOHprofiler_csv_logger::target_problem(int problem_id, int dimension, int instance){
```

Comment:
Please add comments to your functions, and try to make the function names more clear. For example, currently it takes a long time to figure out what target_problem is supposed to do, while this would be much easier with good comments and a more descriptive name.

Reply:
sure, I will add comments later.

Reply:
Thanks, the added comments do really help with understanding the code easier.
Ideally, try to add the comments as soon as you make the function. This helps you know what the function is supposed to do and makes reading and understanding the code a lot quicker.